node/src/test/scala/coop/rchain/node/configuration/commandline/ConfigMapperSpec.scala

```diff
@@ -3,260 +3,235 @@ package coop.rchain.node.configuration.commandline
+        "--protocol-grpc-max-receive-stream-message-length 111111",
+        "--protocol-grpc-stream-chunk-size 111111",
+        "--protocol-max-connections 111111",
+        "--protocol-max-message-consumers 111111",
+        //other vars?
+        "--tls-certificate-path /var/lib/rnode/node.certificate.pem",
+        "--tls-key-path /var/lib/rnode/node.key.pem",
+        "--tls-secure-random-non-blocking",
+        "--api-host localhost",
+        "--api-port-grpc-external 111111",
+        "--api-port-grpc-internal 111111",
+        "--api-port-http 111111",
+        "--api-grpc-max-receive-message-length 111111",
+        "--api-max-blocks-limit 111111",
+        "--api-enable-reporting",
+        "--data-dir /var/lib/rnode",
+        "--lmdb-map-size-rspace 111111",
+        "--lmdb-map-size-blockdagstore 111111",
+        "--lmdb-map-size-blockstore 111111",
+        "--lmdb-map-size-deploystore 111111",
+        // other vars?
+        "--shard-name root",
+        "--fault-tolerance-threshold 111111",
+        "--validator-public-key 111111",
+        "--validator-private-key 111111",
+        "--validator-private-key-path /var/lib/rnode/pem.key",
+        "--casper-loop-interval 111111seconds",
+        "--requested-blocks-timeout 111111seconds",
+        "--finalization-rate 111111",
+        "--max-number-of-parents 111111",
+        "--max-parent-depth 111111",
+        "--fork-choice-stale-threshold 111111seconds",
+        "--fork-choice-check-if-stale-interval 111111seconds",
+        "--synchrony-constraint-threshold 111111",
+        "--height-constraint-threshold 111111",
+        "--frrd-max-peer-queue-size 111111",
+        "--frrd-give-up-after-skipped 111111",
+        "--frrd-drop-peer-after-retries 111111",
+        "--bonds-file /var/lib/rnode/genesis/bonds1.txt",
+        "--wallets-file /var/lib/rnode/genesis/wallets1.txt",
+        "--bond-minimum 111111",
+        "--bond-maximum 111111",
+        "--epoch-length 111111",
+        "--quarantine-length 111111",
+        "--number-of-active-validators 111111",
+        "--deploy-timestamp 111111",
+        "--required-signatures 111111",
+        "--approve-interval 111111seconds",
+        "--approve-duration 111111seconds",
+        "--genesis-validator",
+        "--prometheus",
+        "--influxdb",
+        "--influxdb-udp",
+        "--zipkin",
+        "--sigar"
       ).mkString(" ")
 
     val options = Options(args.split(' '))
-    val config  = ConfigMapper.fromOptions(options)
 
-    val expectedServer =
-      configuration.Server(
+    val defaultConfig = ConfigSource
+      .resources("defaults.conf")
+      .withFallback(
+        ConfigSource.string(
+          s"default-data-dir = /var/lib/rnode"
+        )
+      )
+
+    // Custom reader for PeerNode type
+    implicit val PeerNodeReader = ConfigReader.fromString[PeerNode](
+      ConvertHelpers.catchReadError(s => PeerNode.fromAddress(s).getOrElse(null))
```

Comment:
I believe no, changed to 
https://github.com/rchain/rchain/blob/383f71c62cd2cd331bbb4cac0989fc82217ae4d6/node/src/main/scala/coop/rchain/node/configuration/Configuration.scala#L69-L71

Reply:
This will not propagate the error from `Left(err)` returned from `fromAddress`. There is `ConfigReader.fromStringTry` that is convenient in this case because Either[Throwable, A] is similar to `Try[A]`.
I commented on the third place where similar code is used. :)

Reply:
Ah, understood, will push fix now, there are 3 places where this is used.

Reply:
Fixed